### PR TITLE
SAK-47091 handle spaces in names when sorting

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java [hunk 1]
```diff
@@ -56,6 +57,16 @@ public int compare(User u1, User u2) {
         if (u1 == null) return (nullsLow ? -1 : 1);
         if (u2 == null) return (nullsLow ? 1 : -1);
 
-        return new NullSafeComparator<>(collator, nullsLow).compare(u1.getSortName(), u2.getSortName());
+        // Replace spaces to handle sorting scenarios where surname has space
+        String prop1 = StringUtils.replace(u1.getSortName(), " ", "+");
+        String prop2 = StringUtils.replace(u2.getSortName(), " ", "+");
```
Comment (ern) [inline kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java:62]: Does adjusting the Collator make more sense than a separate String manipulation?

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java [hunk 2]
```diff
@@ -56,6 +57,16 @@ public int compare(User u1, User u2) {
...
+
+        // Secondary sort if full name is identical
+        if (StringUtils.equals(prop1, prop2)) {
```
Comment (plukasew) [inline kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java:65]: This should probably be done with the same NullSafeComparator used above. StringUtils.equals won't ignore case.

**Comment by ottenhoff** [comment]
Where is this line from and who did it?!

String newRules = oldRules.replaceAll("<'\u005f'", "<' '<'\u005f'");

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java (hunk 1 above)
Comment (ottenhoff) [reply kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java:62]: I find this horribly unreadable:

oldRules.replaceAll("<'\u005f'", "<' '<'\u005f'");

Does it make sense to you? is it not working as expected?

**Comment by bjones86** [comment]
Looks like it came from https://github.com/sakaiproject/sakai/commit/91acbaef82a0a53fee7feae90f88cc850123949b

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java (hunk 2 above)
Comment (ottenhoff) [reply kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java]: ok done

**Comment by ottenhoff** [comment]
Maybe 46754fe501f7739efe525f79578c3a7e6d2add2e

**Comment by bjones86** [comment]
Yeah seems likely.

**Comment by ottenhoff** [comment]
the idea of that line seems to be saying that space should come right before this very high unicode char (u005f)

but seems like it doesn't work!

File: kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java (hunk 1 above)
Comment (ern) [reply kernel/kernel-util/src/main/java/org/sakaiproject/util/comparator/UserSortNameComparator.java:57]: yeah you could make the code easier to read

**Review by bjones86 — APPROVED** [review]
LGTM

**Comment by ern** [comment]
Sorry I thought the goal here was to remove those chars!!!
OK sounds fine 👍

**Pull request merged**